src/SimpleMemSystem.cpp

```diff
@@ -228,42 +287,32 @@ SimpleMemSystem::SimpleMemSystem(uint32_t rom_start, uint32_t rom_size, uint32_t
     }
 }
 
-SimpleMemSystem::SimpleMemSystem() : SimpleMemSystem(-1, 0, -1, 0) {}
-
-etiss::int32 SimpleMemSystem::iread(ETISS_CPU *cpu, etiss::uint64 addr, etiss::uint32 len)
+etiss::int32 SimpleMemSystem::iread(ETISS_CPU *, etiss::uint64 addr, etiss::uint32 len)
 {
-    int i_seg = 0;
-    int n_segs = msegs_.size();
-    for (i_seg = 0; n_segs; ++i_seg)
-    {
-        if (msegs_[i_seg]->addr_in_range(addr))
-            break;
-    }
-
-    if (i_seg < n_segs)
-    {
-        return RETURNCODE::NOERROR;
-    }
-
-    if (addr >= rom_start_ && addr < rom_start_ + rom_mem_.size())
-    {
-        return RETURNCODE::NOERROR;
-    }
+    auto it = std::find_if(msegs_.begin(), msegs_.end(), find_fitting_mseg(addr, len));
+    if (it != msegs_.end()) return RETURNCODE::NOERROR;
 
-    AccessError(cpu->instructionPointer, addr, false);
-    return RETURNCODE::IBUS_WRITE_ERROR;
+    std::stringstream ss;
+    ss << "Error during ibus read at address 0x" << std::hex << std::setw(8) << std::setfill('0') << addr << " with length " << len;
+    etiss::log(etiss::ERROR, ss.str());
+    return RETURNCODE::IBUS_READ_ERROR;
 }
 
 etiss::int32 SimpleMemSystem::iwrite(ETISS_CPU *, etiss::uint64 addr, etiss::uint8 *buf, etiss::uint32 len)
 {
-    etiss::log(etiss::VERBOSE, "Blocked instruction write");
+    std::stringstream ss;
+    ss << "Blocked instruction bus write at address 0x" << std::hex << std::setw(8) << std::setfill('0') << addr << " with length" << len;
+    etiss::log(etiss::VERBOSE, ss.str());
```

Comment:
if we return an error code, i think this log should be error level as well

Reply:
Resolved with 2aada2be1e7358baef2949890db3ea8f87419967